client/js/components/header/search.js

```diff
@@ -0,0 +1,58 @@
+import showHide from './../show-hide';
+import { KEYS } from './../../util';
+
+const headerSearch = (el) => {
+  const headerSearchInput = el.querySelector('.js-header-input');
+  const searchToggle = showHide({
+    el: el,
+    activeClass: 'header__form--is-active'
+  });
+
+  const init = () => {
+    handleEvents();
+  };
+
+  const handleEvents = () => {
+    // Only submit the form if the search field is visible and non-empty
+    el.addEventListener('submit', (event) => {
+      if (searchToggle.getActive() && (headerSearchInput.value.trim() !== '')) return;
+
+      event.preventDefault();
+      searchToggle.toggleActive();
+
+      if (searchToggle.getActive()) {
+        searchToggle.drawer.querySelector('input').focus();
+      }
+    });
+
+    // Tabbing backwards or pressing escape on the search field hides the drawer
+    headerSearchInput.addEventListener('keydown', ({ keyCode, shiftKey }) => {
+      if (keyCode !== KEYS.ESCAPE && keyCode !== KEYS.TAB) return;
+      if (keyCode === KEYS.TAB && !shiftKey) return;
+
+      searchToggle.setActive(false);
+      searchToggle.trigger.focus();
+    });
+
+    // Tabbing onto the search button opens the drawer and focuses the search field
+    searchToggle.trigger.addEventListener('keyup', ({ keyCode }) => {
+      if (keyCode !== KEYS.TAB) return;
+      if (searchToggle.getActive()) return;
+
+      searchToggle.setActive(true);
+      searchToggle.drawer.querySelector('input').focus();
+    });
+
+    // Tabbing forwards off the search button closes the drawer
+    searchToggle.trigger.addEventListener('keydown', ({ keyCode, shiftKey }) => {
+      if (keyCode !== KEYS.TAB) return;
+      if (shiftKey) return;
+
+      searchToggle.setActive(false);
+    });
+  };
+
+  init();
+};
+
+export default headerSearch;
```

Comment:
I'm still a little unclear of the logic here - or think it could  be simplified - but let's not block and get it out, maybe when I try it, it'll make more sense.